pkg/e2e/start_stop_test.go

```diff
@@ -222,29 +222,46 @@ func TestStopAlreadyStopped(t *testing.T) {
 }
 
 func TestStartStopMultipleServices(t *testing.T) {
-	cli := NewParallelCLI(t, WithEnv(
-		"COMPOSE_PROJECT_NAME=e2e-start-stop-svc-multiple",
-		"COMPOSE_FILE=./fixtures/start-stop/compose.yaml"))
+	cli := NewParallelCLI(t, WithEnv("COMPOSE_PROJECT_NAME=e2e-start-stop-svc-multiple"))
 	t.Cleanup(func() {
 		cli.RunDockerComposeCmd(t, "down", "--remove-orphans", "-v", "-t", "0")
 	})
 
-	cli.RunDockerComposeCmd(t, "up", "-d", "--wait")
-
-	res := cli.RunDockerComposeCmd(t, "stop", "simple", "another")
-	services := []string{"simple", "another"}
-	for _, svc := range services {
-		stopMsg := fmt.Sprintf("Container e2e-start-stop-svc-multiple-%s-1  Stopped", svc)
-		assert.Assert(t, strings.Contains(res.Stderr(), stopMsg),
-			fmt.Sprintf("Missing stop message for %s\n%s", svc, res.Combined()))
-	}
+	t.Run("starts/stops multiple services", func(t *testing.T) {
+		cli.RunDockerComposeCmd(t, "-f", "./fixtures/start-stop/compose.yaml", "up", "-d", "--wait")
+
+		res := cli.RunDockerComposeCmd(t, "-f", "./fixtures/start-stop/compose.yaml", "stop", "simple", "another")
+		services := []string{"simple", "another"}
+		for _, svc := range services {
+			stopMsg := fmt.Sprintf("Container e2e-start-stop-svc-multiple-%s-1  Stopped", svc)
+			assert.Assert(t, strings.Contains(res.Stderr(), stopMsg),
+				fmt.Sprintf("Missing stop message for %s\n%s", svc, res.Combined()))
+		}
+
+		res = cli.RunDockerComposeCmd(t, "-f", "./fixtures/start-stop/compose.yaml", "start", "simple", "another")
+		for _, svc := range services {
+			startMsg := fmt.Sprintf("Container e2e-start-stop-svc-multiple-%s-1  Started", svc)
+			assert.Assert(t, strings.Contains(res.Stderr(), startMsg),
+				fmt.Sprintf("Missing start message for %s\n%s", svc, res.Combined()))
+		}
+	})
 
-	res = cli.RunDockerComposeCmd(t, "start", "simple", "another")
-	for _, svc := range services {
-		startMsg := fmt.Sprintf("Container e2e-start-stop-svc-multiple-%s-1  Started", svc)
-		assert.Assert(t, strings.Contains(res.Stderr(), startMsg),
-			fmt.Sprintf("Missing start message for %s\n%s", svc, res.Combined()))
-	}
+	t.Run("starts one service out of many, and it's dependencies", func(t *testing.T) {
+		cli.RunDockerComposeCmd(t, "down")
+		cli.RunDockerComposeCmd(t, "-f", "./fixtures/start-stop/start-stop-deps.yaml", "create", "desired")
```

Comment:
Nit: I'd probably make this its own test since it uses a different fixture (which would also help reduce some of the boilerplate of passing `-f ./fixtures/....` everywhere)

(Personally, I think sub-tests in Go are best used for _unit_ tests that have a bunch of cases/permutations - for E2E/more complicated tests, I think it ends up being more confusing/error-prone than helpful, but that's why this is a nit, it's a personal preference 😛)

Reply:
No I agree. I think we could do with big refactors on the E2E side, it's especially confusing to me when subtests are used for setting stuff up/tearing them down and not as self contained tests. But I guess we'll be making changes to our E2E tests :)